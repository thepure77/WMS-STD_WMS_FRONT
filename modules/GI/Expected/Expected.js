(function () {
    'use strict'

    app.component('expected', {
        controllerAs: '$vm',
        templateUrl: "modules/GI/Expected/Expected.html",
        bindings: {
            expected: '=?',
        },
        controller: function ($scope, $filter, $http, /*ngAuthSettings,*/ $state, /*authService*/ pageLoading, $window, commonService, localStorageService, $timeout, $translate, $q, dpMessageBox,expectedFactory,loadTruckItemFactory,scanLoadToTruckFactory) {
            var $vm = this;

            var viewModel = expectedFactory;
            var scanLoadToTruckviewModel = scanLoadToTruckFactory;
            //var loadTruckviewModel = loadTruckItemFactory;
            $scope.isFilter = true;
            var defer = {};
            $scope.filterModel = {};
            $scope.expected = false;
            

            var model = $scope.filterModel;

            $scope.ScanSoNo = function () {
                $vm.filterModel = $vm.filterModel || {};
                $scope.SendSoNo($vm.filterModel).then(function success(res) {
                    $vm.filterModel.pickTicket = res.data[0].pickTicket;
                    setTimeout(() => {
                        var focusElem = jQuery('input[ng-model="$vm.filterModel.cartonQty"]');
                        focusElem[0].focus();

                    }, 200);
                },
                    function error(res) {

                    });
            }
            $scope.SendSoNo = function (model) {
                var deferred = $q.defer();
                viewModel.ScanSoNo(model).then(
                    function success(res) {
                        deferred.resolve(res);
                    },
                    function error(res) {
                        dpMessageBox.alert({
                            ok: 'Close',
                            title: 'Information.',
                            message: "Error!!"
                        })
                    });

                return deferred.promise;
            }

            $scope.ScanCartonQty = function () {
                $vm.filterModel = $vm.filterModel || {};
                $scope.SendcartonQty($vm.filterModel).then(function success(res) {
                    $vm.filterModel.cartNumber = res.data[0].cartNumber;
                },
                    function error(res) {

                    });
            }
            $scope.SendcartonQty = function (model) {
                var deferred = $q.defer();
                viewModel.cartonQty(model).then(
                    function success(res) {

                        deferred.resolve(res);
                    },
                    function error(res) {
                        dpMessageBox.alert({
                            ok: 'Close',
                            title: 'Information.',
                            message: "Error!!"
                        })
                    });

                return deferred.promise;
            }

            $scope.ScanCarton = function () {
                $vm.filterModel = {};
                $vm.filterModel.truckLoadNo = $scope.filterModel.truckLoadNo;
                $vm.filterModel.tagOutNo = $scope.filterModel.tagOutNo;
                $scope.SendCarton($vm.filterModel).then(function success(res) {
                    
                    if(res.data == 'S')
                    {
                        dpMessageBox.alert({
                            ok:'Close',
                            title: 'Information.',
                            message: "Scan carton success!!"
                        })
                    } else if(res.data == 'CartonEmpty')
                    {
                        dpMessageBox.alert({
                            ok:'Close',
                            title: 'Information.',
                            message: "Carton is empty!!"
                        })
                    } else {
                        dpMessageBox.alert({
                            ok:'Close',
                            title: 'Information.',
                            message: "Cannot scan "
                        })
                    }
                    
                    $vm.expected($vm.filterModel);
                    $scope.filterModel.tagOutNo = '';
                },
            
                function error(res) {

                });
                
            }
            $scope.SendCarton = function (model) {
                var deferred = $q.defer();
                scanLoadToTruckviewModel.ScanCarton(model).then(
                    function success(res) {
                        deferred.resolve(res);
                    },
                    function error(res) {
                        dpMessageBox.alert({
                            ok: 'Close',
                            title: 'Information.',
                            message: "Error!!"
                        })
                    });

                return deferred.promise;
            }

            $vm.expected = function (param) {
                
                $scope.filterModel = param;
                var truckLoadNo = $scope.filterModel.truckLoadNo;
                defer = $q.defer();
                $scope.expected = true;
                if (param != undefined) {
                    if(truckLoadNo != undefined)
                    {
                        var model = $scope.filterModel;
                        viewModel.getExpectedRF(truckLoadNo).then(function (res) {
                            $vm.filterModel = res.data;
                            $vm.searchResultModel = res.data;
                        });
                    }

                }
                else {
                    $scope.buttons.add = true;
                    $scope.buttons.update = false;
                }

                return defer.promise;
            };

            $scope.back = function () {
                $state.reload();
                $scope.filterModel = {};
                $scope.expected = false;
                defer.resolve('-1');     
            }

            $scope.buttons = {
                back: true
            };

            $scope.getColor = function (IsCartonEmpty) {
                if (IsCartonEmpty == 1) {
                    //return "background:rgb(220, 20, 60);color:rgb(255, 255, 255)";
                    return "color:rgb(255, 0, 0)";
                }
            }
        }
    })
})();