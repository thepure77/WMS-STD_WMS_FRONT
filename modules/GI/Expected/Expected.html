
<div class="row" ng-show="expected">
        
    <div class="div-shadow">
        <div class="form-group" style="margin-top:15px;">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>{{ 'LOAD_NO' | translate }}</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-12">
                        <input type="text" class="form-control" ng-model="filterModel.truckLoadNo" disabled />
                    </div>
                </div>
            </div>
        </div>
        <div class="form-group">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <!-- <span><a ng-click='$vm.chkCartonList($vm.filterModel)'>Carton</a></span> -->
                        <span>{{ 'CARTON' | translate }}</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-12">
                        <input type="text" class="form-control" ng-model="filterModel.tagOutNo" ks-enter="ScanCarton()"/>
                    </div>
                </div>
            </div>
        </div>
        <div class="form-group" style="margin-top:15px;">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="table-responsive kl-tbl-height table-overflow">
                    <table class="table table-striped table-bordeered table-width-100">
                        <thead>
                            <tr>
                                <th class="text-center">{{ 'SALES_ORDER_NO' | translate }}</th>
                                <th class="text-center">{{ 'PICK_TICKET_NO' | translate }}</th>
                                <th class="text-center">{{ 'ZONE' | translate }}</th>
                                <th class="text-center">{{ 'QTY' | translate }}</th>
                                <th class="text-center">{{ 'COUNT' | translate }}</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr ng-repeat="data in $vm.searchResultModel" ng-if="$vm.searchResultModel.length != 0" style="{{getColor(data.isCartonEmpty)}}">
                                <td class="text-center">{{data.planGoodsIssueNo}}</td>
                                <td class="text-center">{{data.tagOutPickNo}}</td>
                                <td class="text-center">{{data.zoneName}}</td>
                                <td class="text-center">{{data.confirmTagOutQty}}</td>
                                <td class="text-center">{{data.cartonCount}}</td>
                            </tr>
                            <tr ng-if="$vm.searchResultModel.length == 0">
                                <td colspan="4" class="text-center">{{ 'NO_DATA' | translate }}</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
        <!-- <div class="form-group">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>Carton Qty</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-6">
                        <input type="text" class="form-control" ng-model="$vm.filterModel.cartonQty" ks-enter="ScanCartonQty()" disabled/>
                    </div>
                </div>
            </div>
        </div>

        <div class="form-group" style="margin-top:15px;">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>SO No</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-12">
                        <input type="text" class="form-control" ng-model="$vm.filterModel.soNo" ks-enter="ScanSoNo()" disabled/>
                    </div>
                </div>
            </div>
        </div>

        <div class="form-group">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>Carton Qty</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-6">
                        <input type="text" class="form-control" ng-model="$vm.filterModel.cartonQty" ks-enter="ScanCartonQty()" disabled/>
                    </div>
                </div>
            </div>
        </div>
        
        <div class="form-group" style="margin-top:15px;">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>SO No</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-12">
                        <input type="text" class="form-control" ng-model="$vm.filterModel.soNo" ks-enter="ScanSoNo()" disabled/>
                    </div>
                </div>
            </div>
        </div>

        <div class="form-group">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="col-lg-6 col-md-6 col-xs-12">
                    <div class="col-lg-8 col-md-3 col-xs-12">
                        <span>Carton Qty</span>
                    </div>
                    <div class="col-lg-9 col-md-9 col-xs-6">
                        <input type="text" class="form-control" ng-model="$vm.filterModel.cartonQty" ks-enter="ScanCartonQty()" disabled/>
                    </div>
                </div>
            </div>
        </div> -->
        
        
        <div class="form-group">
            <div class="col-lg-12 col-md-12 col-xs-12">
                <div class="text-right padding-m">
                    <button ng-click="back()" class="btn btn-success hidden-xs" ng-show="buttons.back">Back</button>
                    <button ng-click="back()" class="btn btn-success btn-block btn-md visible-xs mt-lg" ng-show="buttons.back">Back</button>
                </div>
            </div>
        </div>
    </div>
</div>